grovedb/src/lib.rs

```diff
@@ -232,8 +231,38 @@ impl GroveDb {
         Err(Error::ReferenceLimit)
     }
 
-    pub fn proof(&self) -> ! {
-        todo!()
+    pub fn proof(&self, path: &[&[u8]], key: &[u8]) -> Result<Vec<Vec<u8>>, Error> {
+        let mut split_path = Some((&key, path));
+        let mut proofs: Vec<Vec<u8>> = Vec::new();
+
+        while let Some((key, path_slice)) = split_path {
+            if path_slice.is_empty() {
+                // Get proof for root tree at current key
+                let root_key_index = self
+                    .root_leaf_keys
+                    .get(*key)
+                    .ok_or(Error::InvalidPath("root key not found"))?;
+                proofs.push(self.root_tree.proof(&vec![*root_key_index]).to_bytes());
```

Comment:
```suggestion
                proofs.push(self.root_tree.proof(&[*root_key_index]).to_bytes());
```

it should work without additional vector allocation